lib/screens/lock/lock_screen.dart

```diff
@@ -0,0 +1,302 @@
+import 'package:deus/core/widgets/header_with_address.dart';
+import 'package:deus/core/widgets/selection_button.dart';
+import 'package:deus/core/widgets/text_field_with_max.dart';
+import 'package:deus/statics/my_colors.dart';
+import 'package:deus/statics/styles.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_countdown_timer/flutter_countdown_timer.dart';
+
+enum LockStates {
+  hasToApprove,
+  pendingApprove,
+  canLock,
+  lockPendingApprove,
+  lockApproved
+}
+
+class LockScreen extends StatefulWidget {
+  static const url = '/Lock';
+
+  @override
+  _LockScreenState createState() => _LockScreenState();
+}
+
+class _LockScreenState extends State<LockScreen> {
+  final String address = '0x3a6dabD6B5C75291a3258C29B418f5805792a87e';
+
+
+  final balance = 13.234;
+
+  final gradient = MyColors.blueToGreenGradient;
+
+  LockStates _lockState = LockStates.hasToApprove;
+  bool _showToast = false;
+
+  final _textController = TextEditingController();
+
+  int endTime = DateTime.now().millisecondsSinceEpoch + 1000 * 30; // current Time in the locking period
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      backgroundColor: Color(MyColors.Background),
+      body: SafeArea(
+        child: SingleChildScrollView(
+          padding: const EdgeInsets.symmetric(vertical: 20, horizontal: 15),
+          child: SizedBox(
+            height: MediaQuery.of(context).size.height + 50,
+            child: Column(
+              mainAxisSize: MainAxisSize.min,
+              crossAxisAlignment: CrossAxisAlignment.start,
+              children: [
+                HeaderWithAddress(
+                  walletAddress: address,
+                ),
+                SizedBox(
+                  height: 23,
+                ),
+                BackButtonWithText(),
+                SizedBox(
+                  height: 20,
+                ),
+                Text(
+                  'Lock your DEA',
+                  style: TextStyle(fontSize: 25),
+                ),
+                SizedBox(
+                  height: 20,
+                ),
+                Text(
+                  'If you lock your DEA you will mint sDEA which can be used for single asset staking or for liquidity pools.',
+                  style: MyStyles.whiteMediumTextStyle,
+                ),
+                CountdownTimer(
+                  endTime: endTime,
+                  widgetBuilder: (_, time) {
+                    if (time == null) {
+                      return Text('Current locking period still lasts 0 Days, 0 Hours, 0 Minutes and 0 Seconds.', style: MyStyles.gradientMediumTextStyle);
+                    } else {
+                      return Text(
+                        'Current locking period still lasts ${time.days != null ? time.days : 0} Days, ${time.hours != null ? time.hours : 0} Hours, ${time.min != null ? time.min : 0} Minutes and ${time.sec != null ? time.sec : 0} Seconds.',
```

Comment:
Bessere Schreibweise ist meistens mit `??`.

- [ ] `time.days != null ? time.days : 0` wird zu `time.days ?? 0`.
- [ ] ebenso mit den anderen Eigenschaften, also hours, min und seconds.

Übrigens: Du könntes theoretisch sogar die erste if Abrage sparen, indem du `time?.days ?? 0` verwendest. Das sagt dann quasi: "Falls `time` nicht null ist, nimm `time.days`. Falls aber eins von den beiden null ist, dann nimm die Zahl 0. 
Wenn man einfach nur `beispielObjekt.eigenschaft ?? 0` benutzen würde (also ohne das erste Fragezeichen hinter beispielObjekt), hätte man ein Problem, wenn beispielObjekt null ist: Dann würde man nen Fehler kriegen, weil `null` nicht das Feld `eigenschaft hat`. 

Da es aber viel übersichtlicher mit der if & else Schreibweise in diesem Beispiel hier ist, also der Code besser lesbar ist, kannst du es dann so lassen wie es ist.

Reply:
Wenn die Erklärung nicht gut genug war, schreib mir auf Slack. Ist mir wichtig, dass du das verstehst und in Zukunft anwenden kannst.